conformance/tests/mesh/httproute-rewrite-path.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package meshtests
+
+import (
+	"testing"
+
+	"sigs.k8s.io/gateway-api/conformance/utils/echo"
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+	"sigs.k8s.io/gateway-api/pkg/features"
+)
+
+func init() {
+	MeshConformanceTests = append(MeshConformanceTests, MeshHTTPRouteRewritePath)
```

Comment:
I am maybe out of the loop here. But this is not the first mesh conformance test, we have a few like conformance/tests/mesh-consumer-route.go. Should those be ported over to be "MeshConformanceTests" instead of "ConformanceTests"?

Reply:
It should - this is a concept PR just to see if we actually want to go down this path. As per the previous comment this came up during kubecon to prioritize get tests faster.

Reply:
resolving this, we reached consensus on slack (x-ref: https://kubernetes.slack.com/archives/CR0H13KGA/p1744390252965819)